### PR TITLE
Move rotating kubelet client certificate to beta.

### PR DESCRIPTION
**Pull request by jcbsmpsn**
```release-note
Release the kubelet client certificate rotation as beta.
```

### CODE REVIEW

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ var defaultKubernetesFeatureGates = map[utilfeature.Feature]utilfeature.FeatureS
 	Accelerators:                                {Default: false, PreRelease: utilfeature.Alpha},
 	TaintBasedEvictions:                         {Default: false, PreRelease: utilfeature.Alpha},
 	RotateKubeletServerCertificate:              {Default: false, PreRelease: utilfeature.Alpha},
-	RotateKubeletClientCertificate:              {Default: false, PreRelease: utilfeature.Alpha},
+	RotateKubeletClientCertificate:              {Default: false, PreRelease: utilfeature.Beta},
```
Comment (ericchiang) [inline pkg/features/kube_features.go]: Shouldn't this default to true?

File: cmd/kubelet/app/options/options.go [hunk 2]
```diff
@@ -203,6 +205,8 @@ func (f *KubeletFlags) AddFlags(fs *pflag.FlagSet) {
 		"If the file specified by --kubeconfig does not exist, the bootstrap kubeconfig is used to request a client certificate from the API server. "+
 		"On success, a kubeconfig file referencing the generated client certificate and key is written to the path specified by --kubeconfig. "+
 		"The client certificate and key file will be stored in the directory pointed by --cert-dir.")
+	fs.BoolVar(&f.RotateCertificates, "rotate-certificates", f.RotateCertificates, "<Warning: Beta feature> Auto rotate the kubelet client certificates by requesting new certificates from the kube-apiserver when the certificate expiration approaches.")
+	fs.MarkHidden("rotate-certificates") // Mark this flag as hidden because it is still in Beta.
```
Comment (ericchiang) [inline cmd/kubelet/app/options/options.go]: Has this pattern been used elsewhere? I don't remember us hiding other flags.

**Comment by ericchiang** [comment]
cc @kubernetes/sig-auth-pr-reviews

File: cmd/kubelet/app/options/options.go (hunk 2 above)
Comment (jcbsmpsn) [inline cmd/kubelet/app/options/options.go]: I don't know of it existing anywhere else. You think just the warning text and not hiding it is the best way to go?

File: pkg/features/kube_features.go (hunk 1 above)
Comment (jcbsmpsn) [inline pkg/features/kube_features.go]: Done.

Comment (ericchiang) [inline pkg/features/kube_features.go]: What's the timeline for GA? I'd imagine most setups will still use static TLS for the foreseeable future and it seems odd to turn it on by default if a cluster isn't using it.

Comment (liggitt) [inline pkg/features/kube_features.go]: what is the behavior if the controller manager doesn't have a signing CA set up? does this degrade gracefully if unable to submit the CSR, or if the submitted CSR is never approved?

Comment (jcbsmpsn) [inline pkg/features/kube_features.go]: I mis-understood the behavior of the signing CA. I'm going to turn the flag to false by default, and put the beta feature to true.

**Comment by ericchiang** [comment]
/assign @dchen1107

For approval.

**Pull request merged**